packages/create-app/templates/default-app/packages/backend/Dockerfile

```diff
@@ -1,13 +1,14 @@
+# syntax = docker/dockerfile:1.0-experimental
 FROM node:12-buster
 
 WORKDIR /usr/src/app
 
 # Copy repo skeleton first, to avoid unnecessary docker cache invalidation.
 # The skeleton contains the package.json of each package in the monorepo,
 # and along with yarn.lock and the root package.json, that's enough to run yarn install.
-ADD yarn.lock package.json skeleton.tar ./
+ADD .npmrc* yarn.lock package.json skeleton.tar ./
 
-RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -rf "$(yarn cache dir)"
+RUN --mount=type=secret,id=NPM_TOKEN NPM_TOKEN=$(cat /run/secrets/NPM_TOKEN) yarn install --frozen-lockfile --production --network-timeout 300000 && rm -rf "$(yarn cache dir)"
```

Comment:
Some documentation about how the `build-image` command needs to look like would be interesting.

Do we want to have this in the default app? It would be good to have this approach documented somewhere but I think that this is a too special case that we shouldn't make it a default option. 

btw: we have a similar problem and currently use the following:
```Dockerfile
# Dockerfile

RUN \
  --mount=type=secret,id=npmrc-global,target=/root/.npmrc \
  --mount=type=secret,id=npmrc-local,target=/usr/src/app/.npmrc \
  npm install -g yarn \
  && yarn install --frozen-lockfile --production && rm -rf "$(yarn cache dir)"
```

```diff
-    "build-image": "yarn workspace backend build-image",
+    "build-image": "DOCKER_BUILDKIT=1 backstage-cli backend:build-image --secret id=npmrc-global,src=$HOME/.npmrc --secret id=npmrc-local,src=$PWD/../../.npmrc",
```

Reply:
The way I understand this mechanism is that we can provide additional arguments to `build-image` -- let me write something up.

Reply:
Maybe the more general approach is to provide a custom Dockerfile, or to provide a list of files to include.

Reply:
I've added this to the help string for `backstage-cli backend:build-image`, not sure if there's a better place where this should be documented -- I can't seem to find any documentation in `/docs` that talks about this.